tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -5650,6 +5650,49 @@ TEST_F(OpConverterTest, ConvertConcat) {
   // TestConvertConcat<DT_INT32>(this);
 }
 
+NodeDef MakeCastNodeDef(DataType input_dtype, DataType output_dtype) {
+  Scope s = Scope::NewRootScope();
+  auto input = ops::Placeholder(s.WithOpName("input"), input_dtype);
+  return ops::Cast(s.WithOpName("my_cast"), input, output_dtype)
+      .operation.node()
+      ->def();
+}
+
+template <DataType IN_DTYPE, DataType OUT_DTYPE>
+void TestConvertCast(OpConverterTest* test) {
+
+  test->Reset();
+
+  // Create cast node.
+  const NodeDef node_def = MakeCastNodeDef(IN_DTYPE, OUT_DTYPE);
+
+  test->AddTestTensor("input", /*dims=*/{256, 256, 3}, /*batch_size=*/1,
+                      /*trt_dtype=*/TfDataTypeToTrt(IN_DTYPE));
+
+  // Convert node
+  TRT_TensorOrWeights output;
+  test->RunValidationAndConversion(node_def);
+
+  // Collect output
+  TF_EXPECT_OK(test->GetTensorOrWeights("my_cast", &output));
+
+  // Verify Result
+  ASSERT_TRUE(output.is_tensor());
+  ExpectTrtDimsEqualsArray({256, 256, 3}, output.tensor()->getDimensions());
+  EXPECT_EQ(TfDataTypeToTrt(OUT_DTYPE), output.tensor()->getType())
+      << "Expected " << DebugString(TfDataTypeToTrt(OUT_DTYPE)) << " vs. actual "
+      << DebugString(output.tensor()->getType());
+}
+
+TEST_F(OpConverterTest, ConvertCast) {
+#if IS_TRT_VERSION_GE(7, 0, 0, 11)
+  TestConvertCast<DT_FLOAT, DT_FLOAT>(this);
```

Comment:
why do we test cast to itself? shall we expect this is const-fold/removed by the grappler passes?

Reply:
It doesn't hurt. If, for any reason, a castto itself is present and not removed, we should be able to convert it since it's supported by TRT